dvc/path_info.py

```diff
@@ -121,11 +121,15 @@ class URLInfo(_BasePath):
     DEFAULT_PORTS = {"http": 80, "https": 443, "ssh": 22, "hdfs": 0}
 
     def __init__(self, url):
-        p = urlparse(url)
-        assert not p.query and not p.params and not p.fragment
-        assert p.password is None
+        if isinstance(url, URLInfo):
+            parts = (url.scheme, url.host, url.user, url.port, url.path)
+        else:
+            p = urlparse(url)
+            assert not p.query and not p.params and not p.fragment
+            assert p.password is None
+            parts = (p.scheme, p.hostname, p.username, p.port, p.path)
 
-        self._fill_parts(p.scheme, p.hostname, p.username, p.port, p.path)
+        self._fill_parts(*parts)
```

Comment:
Oh, actually new logic here. Added this to make `URLInfo(URLInfo())` work the same way as `PathInfo(PathInfo())` and pathlib's `Path(Path())` work. Let me check if we still need it and if we do -- will add a unit test.